File: Data Structures & Algorithms/validate-parentheses/submission-2.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (c == '(' || c == '[' || c == '{') stack.push(c);
            else {
                if (stack.size() == 0) {
                    return false;
                }
                char j = stack.pop();
                if (c == ')')  {
                    if (j != '(') return false;
                }
                if (c == '}')  {
                    if (j != '{') return false;
                }
                if (c == ']')  {
                    if (j != '[') return false;
                }
            }
        }

        if (stack.size() > 0) return false;
        return true;
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for token in tokens:
            if token not in ('+', '-', '*', '/'):
                stack.append(token)
            else:
                a = stack.pop()
                b = stack.pop()
                if token == '+':
                    stack.append(int(b) + int(a))
                if token == '-':
                    stack.append(int(b) - int(a))
                if token == '*':
                    stack.append(int(b) * int(a))
                if token == '/':
                    stack.append(int(b) / int(a))
        return int(stack.pop())